lib/devices/default.js

```diff
@@ -331,7 +331,19 @@ let defaultsgen2 = {
   'hostname': {
     mqtt: {
       http_publish: '/rpc/Shelly.GetStatus',
-      http_publish_funct: async (value) => { return value ? JSON.parse(value).wifi.sta_ip : undefined; }
+      http_publish_funct: async (value) => {
+        if (value) {
+          const valueObj = JSON.parse(value);
+
+          if (valueObj.wifi.sta_ip) {
+            return valueObj.wifi.sta_ip;
+          } else if (valueObj.eth.ip) {
+            return valueObj.eth.ip;
+          }
```

Comment:
Would it make more sense to swap the conditions to prefer Ethernet over Wifi? Or should it be up to the user to disable Wifi, if a Shelly is connected over Ethernet?

Reply:
At the moment this is just an information. The adapter doesn't know if the device is connected via ethernet or wifi.